src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/common/jpa/ExtraJpaConfiguration.kt

```diff
@@ -0,0 +1,25 @@
+package uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.common.jpa
+
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.data.domain.AuditorAware
+import org.springframework.data.jpa.repository.config.EnableJpaAuditing
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.core.userdetails.UserDetails
+import java.util.Optional
+
+@Configuration
+@EnableJpaAuditing(modifyOnCreate = false)
+class ExtraJpaConfiguration {
+  @Bean
+  fun auditorAware() = AuditorAware {
+    Optional.ofNullable(
+      when (val principal = SecurityContextHolder.getContext().authentication?.principal) {
+        is String -> principal
+        is UserDetails -> principal.username
```

Comment:
Out of interest, do we get the user's first and last name on the `principal` here? I'm now thinking the full name would be more user-friendly than the username (which often looks like `EG012345` in prod). If not, I'm assuming we'd need to make a request to the `manage-users-api` to fetch that?